pkg/signal/trap_linux_test.go

```diff
@@ -0,0 +1,66 @@
+// +build linux
+
+package signal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrap(t *testing.T) {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+	for k, v := range sigmap {
+		wd, _ := os.Getwd()
+		testHelperCode := wd + "/testfiles/main.go"
+		cmd := exec.Command("go", "build", "-o", "main", testHelperCode)
+		err := cmd.Run()
+		require.NoError(t, err)
+		cmd = exec.Command("./main")
+		cmd.Env = append(os.Environ(), fmt.Sprintf("SIGNAL_TYPE=%s", k))
+
+		err = cmd.Start()
+		require.NoError(t, err)
+		err = cmd.Wait()
+		if e, ok := err.(*exec.ExitError); ok {
+			code := e.Sys().(syscall.WaitStatus).ExitStatus()
+			switch k {
+			case "TERM", "QUIT":
+				assert.Equal(t, (128 + int(v.(syscall.Signal))), code)
+			case "INT":
+				assert.Equal(t, 99, code)
+			}
+			continue
+		}
+		t.Fatal("process didn't end with any error")
+	}
+}
+func TestDumpStacks(t *testing.T) {
+	directory, errorDir := ioutil.TempDir("", "test")
```

Comment:
A couple small things
* maybe `test-dump-tasks` instead of just `test` here, in case the directory is accidentally left over somehow (os.RemoveAll() fails for example) we'll be able to find the source of the directory with a more unique name
* `errorDir` makes it sound like a directory, could this be just `err` ?